src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -163,6 +168,76 @@ protected ExecutionStrategy(DataFetcherExceptionHandler dataFetcherExceptionHand
      */
     public abstract CompletableFuture<ExecutionResult> execute(ExecutionContext executionContext, ExecutionStrategyParameters parameters) throws NonNullableFieldWasNullException;
 
+    public CompletableFuture<Object> executeObject(ExecutionContext executionContext, ExecutionStrategyParameters parameters) throws NonNullableFieldWasNullException {
```

Comment:
executeObject is really what we are doing when we call back to  the ExecutionStrategy recursively